src/field/field3d.cxx

```diff
@@ -150,54 +135,53 @@ Field3D* Field3D::timeDeriv() {
 void Field3D::splitYupYdown() {
   TRACE("Field3D::splitYupYdown");
   
-  if((yup_field != this) && (yup_field != nullptr))
+  if (!yup_fields.empty()) {
```

Comment:
Should we also check `ydown_fields`?

Reply:
Possibly? They are private and should always be in sync, so I think we would have problems if only one was empty.

Reply:
We could add an `ASSERT1(ydown_fields.empty())` just to check?

Reply:
Yes, that's sensible. Probably in `mergeYupYdown` too